### PR TITLE
Nitishm/enhancement/42/module config concurrency param

### LINKED ISSUE #42: Support setting maximum concurrent requests for the HTTP library
There is currently no support for setting the maximum number of concurrent requests a guest module can perform.
Since the HTTP library supports it, we should expose it as a field in the module config as well:

https://github.com/deislabs/wasi-experimental-http/blob/6a6226ead5ae2335ccde0a7e9a0b4a128d46249f/crates/wasi-experimental-http-wasmtime/src/lib.rs#L296-L298

### PR DESCRIPTION
**Pull request by nitishm**
Closes #42

### CODE REVIEW

**Comment by nitishm** [comment]
@thomastaylor312 I need to test - you can take a look whether or not superficially it looks right

File: src/lib.rs [hunk 1]
```diff
@@ -31,6 +31,7 @@ pub struct Router {
     cache_config_path: PathBuf,
     module_cache: PathBuf,
     default_host: String,
+    http_concurrency: Option<u32>,
```
Comment (radu-matei) [inline src/lib.rs:34]: Could this be `http_max_concurrency`, or something a bit more explicit?

File: src/lib.rs [hunk 2]
```diff
@@ -44,6 +45,7 @@ impl Router {
             module_cache_dir: PathBuf::default(),
             base_log_dir: PathBuf::default(),
             default_host: String::default(),
+            http_concurrency: Option::None,
```
Comment (thomastaylor312) [inline src/lib.rs:48]: `None` is imported by default in rust, so this can just be `http_concurrency: None`

File: src/lib.rs (hunk 1 above)
Comment (thomastaylor312) [reply src/lib.rs:34]: I agree here. This should be more specific

File: src/lib.rs [hunk 3]
```diff
@@ -133,6 +141,7 @@ impl Default for RouterBuilder {
                 .map(|td| td.into_path())
                 .unwrap_or_default(),
             default_host: String::from("localhost:3000"),
+            http_concurrency: Some(1),
```
Comment (thomastaylor312) [inline src/lib.rs:144]: Because this is an option, it should probably be `None` as the default we unwrap to is `1`

File: src/lib.rs [hunk 4]
```diff
@@ -316,6 +329,7 @@ impl ModuleConfig {
         cache_config_path: &Path,
         module_cache_dir: &Path,
         base_log_dir: &Path,
+        http_concurrency: &Option<u32>,
```
Comment (thomastaylor312) [inline src/lib.rs:332]: Nit: I don't think there is a need to have this be borrowed. `u32` is `Copy`, so there isn't any ownership issues here for someone passing in something like `self.http_concurrency`. If you do want to keep the borrow, I'd recommend `Option<&u32>` over a borrowed option, but even that looks a bit weird to me (meaning I haven't seen that anywhere in Rust before)

File: src/runtime/mod.rs [hunk 5]
```diff
@@ -983,8 +979,8 @@ mod test {
 
         let log_tempdir = tempfile::tempdir().expect("Unable to create tempdir");
         let cache_tempdir = tempfile::tempdir().expect("new cache temp dir");
-
-        mc.build_registry(&cache, cache_tempdir.path(), log_tempdir.path())
+        let http_conc = None;
```
Comment (thomastaylor312) [inline src/runtime/mod.rs:982]: Nit: You should be able to skip this and just pass `&None` on the next line.

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (nitishm) [reply src/lib.rs]: done

File: src/lib.rs (hunk 2 above)
Comment (nitishm) [reply src/lib.rs]: done

File: src/lib.rs (hunk 3 above)
Comment (nitishm) [reply src/lib.rs]: done

File: src/lib.rs (hunk 4 above)
Comment (nitishm) [reply src/lib.rs]: done

File: src/runtime/mod.rs (hunk 5 above)
Comment (nitishm) [reply src/runtime/mod.rs]: done

**Review by thomastaylor312 — APPROVED** [review]
Good to go once rebased!

**Pull request merged**